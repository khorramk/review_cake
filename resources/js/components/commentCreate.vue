<template>
    <div class="comment-create__container">
        <form @submit.prevent="addComments" class="comment-create__container__create-comment-form" >
            <textarea class="comment-create__container__create-comment-form__body" name="comment" id="" cols="30" rows="10"  v-model="commentBody"></textarea>
            <button :disabled="isDisable" class="comment-create__container__create-comment-form__submit-btn" name="Submit" type="submit" value="add comments">Submit</button>
        </form>
    </div>
</template>
<script>
export default {
    data(){
        return {
            commentBody: '',
            isDisable: false
        }
    },
    props: {
        reviewId: {
            default: null,
            type: Number
        },
    },
    methods: {
        addComments(){
        this.isDisable = true;
        axios.post('/api/comments', {
            'reviewId': this.$props.reviewId,
            'commentBody': this.commentBody
        }).then(()=> {
            window.location.href = '/api/reviews';
        });  
        }
    }
};
</script>

