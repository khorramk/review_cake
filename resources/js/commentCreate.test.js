//test.js

import Vue from 'vue';
import axios from 'axios';
import VueRouter from 'vue-router';
import commentCreate from './components/commentCreate';

const vm = new Vue(commentCreate);

it('renders the elements', () => {
   
    vm.$nextTick().then(()=> {
 
     expect(vm).toMatchSnapshot();
    }).catch().finally()
  })