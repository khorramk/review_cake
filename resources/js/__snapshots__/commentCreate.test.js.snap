// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the elements 1`] = `
Vue {
  "$attrs": Object {},
  "$children": Array [],
  "$createElement": [Function],
  "$listeners": Object {},
  "$options": Object {
    "_base": [Function],
    "components": Object {},
    "data": [Function],
    "directives": Object {},
    "filters": Object {},
    "methods": Object {
      "addComments": [Function],
    },
    "render": [Function],
    "staticRenderFns": Array [],
  },
  "$parent": undefined,
  "$refs": Object {},
  "$root": [Circular],
  "$scopedSlots": Object {},
  "$slots": Object {},
  "$vnode": undefined,
  "_c": [Function],
  "_data": Object {
    "commentBody": "",
    "isDisable": false,
  },
  "_directInactive": false,
  "_events": Object {
    "submit": Array [
      [Function],
    ],
  },
  "_hasHookEvent": false,
  "_inactive": null,
  "_isBeingDestroyed": false,
  "_isDestroyed": false,
  "_isMounted": false,
  "_isVue": true,
  "_renderProxy": Vue {
    "$attrs": Object {},
    "$children": Array [],
    "$createElement": [Function],
    "$listeners": Object {},
    "$options": Object {
      "_base": [Function],
      "components": Object {},
      "data": [Function],
      "directives": Object {},
      "filters": Object {},
      "methods": Object {
        "addComments": [Function],
      },
      "render": [Function],
      "staticRenderFns": Array [],
    },
    "$parent": undefined,
    "$refs": Object {},
    "$root": [Circular],
    "$scopedSlots": Object {},
    "$slots": Object {},
    "$vnode": undefined,
    "_c": [Function],
    "_data": Object {
      "commentBody": "",
      "isDisable": false,
    },
    "_directInactive": false,
    "_events": Object {
      "submit": Array [
        [Function],
      ],
    },
    "_hasHookEvent": false,
    "_inactive": null,
    "_isBeingDestroyed": false,
    "_isDestroyed": false,
    "_isMounted": false,
    "_isVue": true,
    "_renderProxy": [Circular],
    "_self": [Circular],
    "_staticTrees": null,
    "_uid": 0,
    "_vnode": null,
    "_watcher": null,
    "_watchers": Array [],
    "addComments": [Function],
    "commentBody": "",
    "isDisable": false,
  },
  "_self": [Circular],
  "_staticTrees": null,
  "_uid": 0,
  "_vnode": null,
  "_watcher": null,
  "_watchers": Array [],
  "addComments": [Function],
  "commentBody": "",
  "isDisable": false,
}
`;
